### PR TITLE
Destroy gardener-resource-manager early in the shoot deletion flow

### LINKED ISSUE #5464: The high availabilty of gardener resource manager should be disabled right after the kube-apiserver is destroyed
**How to categorize this issue?**

/area ops-productivity robustness
/kind enhancement

**What would you like to be added**:
The resource manager to be destroyed right after the kube-apiserver in deletion and hibernation flow. Or at least its pod disruption budget to be deleted.

**Why is this needed**:
Once the kube-apiserver of a shoot cluster is no longer deployed all replicas of the gardener resource manager are no longer working as they cannot connect to the shoot cluster. Due to this and the PDB, none of the replicas can be gracefully deleted. Most of the time this is not a problem, but lifecycle operation with the seed clusters are unnecessary prolonged until a forceful drain of the nodes is applied. Depending on other settings and home much shoots with failed resource managed are deployed, an average sized seed cluster might not complete its node roll out in 12 hours, while it was taking 2-3 hours previously.

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement 

**What this PR does / why we need it**:
Destroy gardener-resource-manager early in the shoot deletion flow.
In the hibernation flow the GRM is already scaled down to 0 replicas, so no changes needed there.

**Which issue(s) this PR fixes**:
Fixes #5464 

**Special notes for your reviewer**:
@plkokanov I will be grateful if you include this in the v1.41 milestone.

**Release note**:

```other operator
The `gardener-resource-manager` is now destroyed early in the deletion flow. In this way there will not be a failing deployment with PDB that prevents graceful termination of seed nodes.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/gardenlet/controller/shoot/shoot_control_delete.go [hunk 1]
```diff
@@ -545,6 +545,11 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerService.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(deleteKubeAPIServer, destroyKubeAPIServerSNI),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying gardener-resource-manager",
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.ResourceManager.Destroy),
+			Dependencies: flow.NewTaskIDs(deleteKubeAPIServer),
+		})
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: Wait a sec, doesn't the `Deleting referenced resources` step require gardener-resource-manager to be able to delete the `referenced-resources` ManagedResource? If you destroy GRM here, will the deletion of the `referenced-resources` MR succeed?

Comment (vpnachev) [reply pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: /hold
until the above is sorted out

Comment (rfranzke) [reply pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: Too many non-obvious cases :) Probably we should extend the integration test with using referenced resources to cover this case?

Comment (ialidzhikov) [reply pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: Ah, I see that `referenced-resources` MR is with class `seed`. So I think the central GRM should take care of its deletion.

Comment (ialidzhikov) [reply pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: /unhold

Comment (vpnachev) [reply pkg/gardenlet/controller/shoot/shoot_control_delete.go:552]: Just looking at the flow, the dependency graph is:
```
destroyGardenerResourceManager <- deleteKubeAPIServer <- syncPointCleaned <- waitUntilManagedResourcesDeleted
```

I also had the chance today to test the change and indeed the only managed resource left is `referenced-resources` but it is managed by the seed resource manager.

**Review by ialidzhikov — APPROVED** [review]
Thanks!

/lgtm

**Comment by vpnachev** [comment]
Friendly ping to @plkokanov to take a look and check if this can be included in the v1.41 milestone.

**Pull request merged**